lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -352,7 +352,7 @@ def self.extract_prospect_from_response(prospect_node, fields)
   # @param [Array<Hash>] prospects an array of prospect data
   # @return [String] a URL
   def self.build_batch_url(api_endpoint, prospects)
-    prospects_payload_json_encoded = URI.encode({prospects: prospects}.to_json)
+    prospects_payload_json_encoded = URI::DEFAULT_PARSER.escape({prospects: prospects}.to_json)
```

Comment:
Would `CGI.escape` work here?

Reply:
I don't think so, no; `CGI.escape` escapes some characters which `URI.encode` does not:

```ruby
[development] dashboard > prospects = [{foo: 'bar'}]
=> [{:foo=>"bar"}]
[development] dashboard > URI.encode({prospects: prospects}.to_json)
(irb):2: warning: URI.escape is obsolete
=> "%7B%22prospects%22:[%7B%22foo%22:%22bar%22%7D]%7D"
[development] dashboard > CGI.escape({prospects: prospects}.to_json)
=> "%7B%22prospects%22%3A%5B%7B%22foo%22%3A%22bar%22%7D%5D%7D"
[development] dashboard > URI::DEFAULT_PARSER.escape({prospects: prospects}.to_json)
=> "%7B%22prospects%22:[%7B%22foo%22:%22bar%22%7D]%7D"
```

It's of course possible that pardot can deal with these escaped characters, but I couldn't figure out a good way to test that so I decided it was best to exactly preserve existing behavior here, even if that means the end result isn't as lightweight as it could be.